sw/device/tests/keymgr_sideload_otbn_test.c

```diff
@@ -83,6 +84,38 @@ static bool run_x25519_app(otbn_t *otbn_ctx, uint32_t *result) {
   return true;
 }
 
+static bool run_x25519_app_clearkey(otbn_t *otbn_ctx, uint32_t *result) {
+  // Copy the input argument (Montgomery u-coordinate).
+  CHECK(otbn_copy_data_to_otbn(otbn_ctx, sizeof(encoded_u), &encoded_u,
+                               kOtbnVarEncU) == kOtbnOk);
+
+  // Run the OTBN program and wait for it to complete.
+  LOG_INFO("Starting OTBN program...");
+  CHECK(otbn_execute(otbn_ctx) == kOtbnOk);
+  if (otbn_busy_wait_for_done(otbn_ctx) != kOtbnOk) {
+    // Error during execution; retrieve error bits and instruction count to
+    // help with debugging and then exit early.
+    dif_otbn_err_bits_t err_bits;
+
+    CHECK(dif_otbn_get_err_bits(&otbn_ctx->dif, &err_bits) == kDifOk);
+    uint32_t insn_count;
+    CHECK(dif_otbn_get_insn_cnt(&otbn_ctx->dif, &insn_count) == kDifOk);
+    LOG_INFO(
+        "OTBN encountered an invalid key error.\n  Error bits: 0x%08x\n  "
+        "Instruction "
+        "count: 0x%08x",
+        err_bits, insn_count);
+    CHECK(err_bits == invalid_key_err);
+    return false;
+  }
+
+  // Copy the result (also a 256-bit Montgomery u-coordinate).
+  CHECK(otbn_copy_data_from_otbn(otbn_ctx, 32, kOtbnVarEncResult, result) ==
+        kOtbnOk);
+
+  return true;
+}
```

Comment:
can we just modify `run_x25519_app` to take the expect err_bits as input and compare it, so that we reduce some duplication?